great_expectations/datasource/fluent/interfaces.py

```diff
@@ -1085,7 +1085,9 @@ def _validator(self) -> V1Validator:
                 "We can't validate batches that are attached to datasources without a data context"
             )
         batch_definition = self.data_asset.add_batch_definition(
-            name="-".join([self.datasource.name, self.data_asset.name, str(uuid.uuid4())])
+            name="-".join([self.datasource.name, self.data_asset.name, str(uuid.uuid4())]),
+            partitioner=self.batch_request.partitioner,
+            batching_regex=self.batch_request.batching_regex,
```

Comment:
I didn't realize the batching_regex was separate from the partitioner in the batch request.

Reply:
I think that will change with @joshua-stauffer 's work

Reply:
fyi Josh's work has merged, so this PR is now updated to reflect that batching regex is just nested in sql partitioners